interfaces/cython/cantera/base.pyx

```diff
@@ -9,6 +9,31 @@ cdef class _SolutionBase:
                   source=None, yaml=None, thermo=None, species=(),
                   kinetics=None, reactions=(), **kwargs):
 
+        # run instantiation only if valid sources are specified
+        if origin or infile or source or yaml or (thermo and species):
+
+            self._cinit(infile=infile, name=name, adjacent=adjacent,
+                        origin=origin, source=source, yaml=yaml,
+                        thermo=thermo, species=species, kinetics=kinetics,
+                        reactions=reactions, **kwargs)
+            return
+
+        self._base = CxxNewSolution()
+        self.base = self._base.get()
```

Comment:
I think this initialization logic misses out on some of the error checking that happened in the old `__cinit__` (and I guess is still there but inaccessible). Ideally, the _only_ new cases allowed by this PR would be:
```python
ct.Solution()
ct.ThermoPhase()
ct.Kinetics()
ct.Transport()
```
But right now, you can do things like
```python
ct.ThermoPhase(thermo='ideal-gas')
ct.Kinetics(kinetics='gas')
```
Both of which still just return objects of the base class. Previously, this would have raised an exception, namely
```
ValueError: Arguments are insufficient to define a phase
```

Reply:
Another good catch. I decided to not allow for 
```
```